x-pack/plugins/cases/public/common/lib/kibana/hooks.ts

```diff
@@ -0,0 +1,134 @@
+
+export const useToasts = (): StartServices['notifications']['toasts'] =>
+  useKibana().services.notifications.toasts;
+
+export const useHttp = (): StartServices['http'] => useKibana().services.http;
+
+interface UserRealm {
+  name: string;
+  type: string;
+}
+
+export interface AuthenticatedElasticUser {
+  username: string;
+  email: string;
+  fullName: string;
+  roles: string[];
+  enabled: boolean;
+  metadata?: {
+    _reserved: boolean;
+  };
+  authenticationRealm: UserRealm;
+  lookupRealm: UserRealm;
+  authenticationProvider: string;
+}
+
+export const useCurrentUser = (): AuthenticatedElasticUser | null => {
+  const [user, setUser] = useState<AuthenticatedElasticUser | null>(null);
+
+  const [, dispatchToaster] = useStateToaster();
+
+  const { security } = useKibana().services;
+
+  const fetchUser = useCallback(() => {
+    let didCancel = false;
+    const fetchData = async () => {
+      try {
+        if (security != null) {
+          const response = await security.authc.getCurrentUser();
+          if (!didCancel) {
+            setUser(convertToCamelCase<AuthenticatedUser, AuthenticatedElasticUser>(response));
+          }
+        } else {
+          setUser({
+            username: i18n.translate('xpack.cases.getCurrentUser.unknownUser', {
+              defaultMessage: 'Unknown',
+            }),
+            email: '',
+            fullName: '',
+            roles: [],
+            enabled: false,
+            authenticationRealm: { name: '', type: '' },
+            lookupRealm: { name: '', type: '' },
+            authenticationProvider: '',
+          });
+        }
+      } catch (error) {
+        if (!didCancel) {
+          errorToToaster({
+            title: i18n.translate('xpack.cases.getCurrentUser.Error', {
+              defaultMessage: 'Error getting user',
+            }),
+            error: error.body && error.body.message ? new Error(error.body.message) : error,
+            dispatchToaster,
+          });
+          setUser(null);
+        }
+      }
+    };
+    fetchData();
+    return () => {
+      didCancel = true;
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [security]);
+
+  useEffect(() => {
+    fetchUser();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+  return user;
+};
+
+export interface UseGetUserSavedObjectPermissions {
+  crud: boolean;
+  read: boolean;
+}
+
+export const useGetUserSavedObjectPermissions = () => {
+  const [
+    savedObjectsPermissions,
+    setSavedObjectsPermissions,
+  ] = useState<UseGetUserSavedObjectPermissions | null>(null);
+  const uiCapabilities = useKibana().services.application.capabilities;
+
+  useEffect(() => {
+    const capabilitiesCanUserCRUD: boolean =
+      typeof uiCapabilities.siem.crud === 'boolean' ? uiCapabilities.siem.crud : false;
+    const capabilitiesCanUserRead: boolean =
+      typeof uiCapabilities.siem.show === 'boolean' ? uiCapabilities.siem.show : false;
```

Comment:
these should not be `siem` anymore, @XavierM can you please advise how to handle?

Reply:
We need to talk about this one?, we can keep it like that for now. But we need to have a plan an d I am going to remove it when I am taking care of the privileges

Reply:
Yea agreed. Just to make sure I'm understanding right...I was trying to trace where these are set, and I believe it comes from registering the plugin with `core.applications.register` in the `plugin.tsx` for `security_solution`?. This should pull the information from the `KibanaContext` in `security_solution` as is right now when the component is consumed there then?